pennylane/interfaces/execution.py

```diff
@@ -382,34 +396,47 @@ def cost_fn(params, x):
         cache = LRUCache(maxsize=cachesize)
         setattr(cache, "_persistent_cache", False)
 
-    batch_execute = set_shots(device, override_shots)(device.batch_execute)
+    if new_device_interface:
+        batch_execute = device.execute
+    else:
+        batch_execute = set_shots(device, override_shots)(device.batch_execute)
 
     if expand_fn == "device":
-        expand_fn = lambda tape: device.expand_fn(tape, max_expansion=max_expansion)
+        if new_device_interface:
 
-    if gradient_fn is None:
-        # don't unwrap if it's an interface device
-        if "passthru_interface" in device.capabilities() or device.short_name == "default.mixed":
-            return batch_fn(
-                qml.interfaces.cache_execute(
-                    batch_execute, cache, return_tuple=False, expand_fn=expand_fn
-                )(tapes)
-            )
-        unwrapped_tapes = tuple(qml.transforms.convert_to_numpy_parameters(t) for t in tapes)
-        res = qml.interfaces.cache_execute(
-            batch_execute, cache, return_tuple=False, expand_fn=expand_fn
-        )(unwrapped_tapes)
+            def expand_fn(tape):  # pylint: disable=function-redefined
+                """A blank expansion function since the new device handles expansion in preprocessing."""
+                return tape
 
-        return batch_fn(res)
+        else:
 
-    if gradient_fn == "backprop" or interface is None:
-        return batch_fn(
-            qml.interfaces.cache_execute(
-                batch_execute, cache, return_tuple=False, expand_fn=expand_fn
-            )(tapes)
+            def expand_fn(tape):  # pylint: disable=function-redefined
+                """A wrapper around the device ``expand_fn``."""
+                return device.expand_fn(tape, max_expansion=max_expansion)
+
+    if gradient_fn in {None, "backprop"} or interface is None:
+        if not (
+            new_device_interface
+            or interface is None
+            or gradient_fn == "backprop"
+            or device.short_name == "default.mixed"
+            or "passthru_interface" in device.capabilities()

```

Comment:
What is happening there? I don't understand why it is added for the old devices?

Reply:
I just simplified and combined the logic from both the `if gradient_fn is None` and `if gradient_fn == "backprop" or interface is None` logical blocks.  They were all basically doing the same thing.  

All the logical branches were basically doing the same thing: caching the execute function, executing the cached execute function, and post-processing the results.  Some logical branches unwrapped the parameters to numpy first and others didn't.  This helps separate more clearly the cases where we don't need to unwrap the parameters to numpy.